src/lib/xpubs.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ @see https://github.com/blockkeeper/blockkeeper-frontend-web/issues/38
+ */
+import { UtxoAccountType } from '@shapeshiftoss/types'
+import { decode, encode } from 'bs58check'
+
+/* ypub and zpub are not standard, it is an extension of the bip32 created by Trezor team
+ * and adopted by the community. Currently, it has a pretty wide support in multiple wallets
+ * including electrum. The only difference comparing to xpub is a prefix, but as it is
+ * a base58 encoded string with a checksum, checksum is also different
+ *
+ * The easiest way to fix it is to decode from base58check, replace the prefix to
+ * standard xpub or tpub and then to encode back to base58check. Then one can use this xpub
+ * as normal bip32 master key.
+ *
+ * It may make sense to remember the type of the public key as it tells what type of script
+ * is used in the wallet.
+ *
+ */
+
+enum PublicKeyType {
```

Comment:
nitpick: this is a bit confusing because it isn't really the type of the public key. BIP44/49/84 keys are all the same once decoded, they're just serialized using different formats. I'd suggest a name like `BIP32VersionBytes`.